sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -453,7 +534,11 @@ class SessionCatalog(
       val db = formatDatabaseName(name.database.getOrElse(currentDb))
       val table = formatTableName(name.table)
       val relationAlias = alias.getOrElse(table)
-      if (name.database.isDefined || !tempTables.contains(table)) {
+      if (db == globalTempViewManager.database) {
+        globalTempViewManager.get(table).map { viewDef =>
+          SubqueryAlias(relationAlias, viewDef, Some(name))
+        }.getOrElse(throw new NoSuchTableException(db, table))
```

Comment:
We will never hit `NoSuchTableException` in the else branch, because we have `!tempTables.contains(table)` in the else if branch.

I don't like this style and prefer the new atomic way which is used for the global temp view, so that we don't need to wrap the whole method in `synchronized` block. But I wanna minimal the changes in this PR and leave this code unchanged. We can improve it in follow-ups.